sound/soc/sof/intel/hda-dsp.c

```diff
@@ -23,7 +23,7 @@
 #include "hda.h"
```

Comment:
@keyonjie @mrajwa Patch is good and I see a length discussion in https://github.com/thesofproject/linux/pull/2184/commits/765492ec2a28f2a5735edcf7c4955a0be1df7d1c . But, but, now the commit message would still need some polish before we send this out. It seems the commit ("ASoC: SOF: Intel: hda: Allow trace DMA in S0 when DSP is in D0I3 for debug") was incorrect. If CONFIG_SND_SOC_SOF_DEBUG is disabled, this patch now enables debugging by default. I.e. by enabling DEBUG option, the only thing you can do more, is to _disable_ traces when system stays in S0. This does not seem right... should we rather drop the option? @ranj063 ?

Reply:
@kv2019i good point about the condition about if CONFIG_SND_SOC_SOF_DEBUG is enabled or not. So, the condition to enable DMA Trace is SND_SOC_SOF_DEBUG_ENABLE_FIRMWARE_TRACE is selected ([1]sdev->dtrace_is_supported && [2]hda_enable_trace_D0I3_S0 && [3]suspend_target == S0). The code should be changed to be this:

```
		if (!sdev->dtrace_is_supported ||
		    !hda_enable_trace_D0I3_S0 ||
		    sdev->system_suspend_target != SOF_SUSPEND_NONE)
			flags = HDA_PM_NO_DMA_TRACE;
```

About the commit message, "ASoC: SOF: Intel: hda: Allow trace DMA in *S0* (condition [3] above) when *DSP is in D0I3* (condition [2] above, by default) for *debug* (condition [1] above)".